model/strand.rb

```diff
@@ -38,7 +38,7 @@ def self.prog_verify(prog)
   def self.lease(id)
     affected = DB[<<SQL, id].first
 UPDATE strand
-SET lease = now() + '120 seconds', schedule = now()
+SET lease = now() + '120 seconds', try = try + 1, schedule = (now() + least(2 ^ least(try, 20) * random(), 600) * '1 second'::interval)
```

Comment:
Shouldn't we put random() to outside of least?
```
now() + least(2 ^ least(try, 20), 600) * random() * '1 second'::interval
```
Currently when try is high, it is very likely that 600 will be picked, causing multiple jobs to be scheduled at the same time. For example if try is 10 we will pick 600 approximately 40% of the time, for 11 it is 70%, for 12 it is 85% and so on.

Reply:
Good idea